Geometry/GEMGeometryBuilder/src/GEMGeometryBuilder.cc

```diff
@@ -93,9 +96,18 @@ void GEMGeometryBuilder::build(GEMGeometry& theGeometry,
     // currently there is no superchamber in the geometry
     // only 2 chambers are present separated by a gap.
     // making superchamber out of the first chamber layer including the gap between chambers
-    if (detIdCh.layer() == 1) {  // only make superChambers when doing layer 1
+
+    // In Run 3 we also have a GE2/1 station at layer 2. We make sure
+    // the superchamber gets built but also we build on the first
+    // layer for the other stations so the superchamber is in the
+    // right position there.
+    if ((seen.find(detIdCh.superChamberId()) == seen.end()) ||
+        detIdCh.layer() < seen[detIdCh.superChamberId()].layer()) {
+      seen[detIdCh.superChamberId()] = detIdCh.chamberId();
+      if (superChambers.find(detIdCh.superChamberId()) != superChambers.end())
+        delete superChambers[detIdCh.superChamberId()];
```

Comment:
@watson-ij If I understand this new code, it builds some super chambers and then deletes them if they are not needed. That seems rather wasteful. There should be a way to build only the correct super chambers.

Reply:
Yes, I had issues with the layer ordering going through the geometry which meant I couldn't assume the first layer is in a particular position. I suppose I could try to cache the fv instead of building here and then build everything outside the loop. I will have to try this tomorrow as our local cluster is down today.